tensorbay/opendataset/nuScenes/loader.py

```diff
@@ -0,0 +1,303 @@
+    "cycle": {"name": "cycle_rider"},
+    "pedestrian": {"name": "pedestrian_motion"},
+}
+
+
+def nuScenes(path: str) -> FusionDataset:  # pylint: disable=invalid-name
+    """Dataloader of the `nuScenes`_ dataset.
+
+    .. _nuScenes: https://www.nuscenes.org/
+
+    The file structure should be like::
+
+        <path>
+            v1.0-mini/
+                maps/
+                    36092f0b03a857c6a3403e25b4b7aab3.png
+                    ...
+                samples/
+                    CAM_BACK/
+                    CAM_BACK_LEFT/
+                    CAM_BACK_RIGHT/
+                    CAM_FRONT/
+                    CAM_FRONT_LEFT/
+                    CAM_FRONT_RIGHT/
+                    LIDAR_TOP/
+                    RADAR_BACK_LEFT/
+                    RADAR_BACK_RIGHT/
+                    RADAR_FRONT/
+                    RADAR_FRONT_LEFT/
+                    RADAR_FRONT_RIGHT/
+                sweeps/
+                    CAM_BACK/
+                    CAM_BACK_LEFT/
+                    CAM_BACK_RIGHT/
+                    CAM_FRONT/
+                    CAM_FRONT_LEFT/
+                    CAM_FRONT_RIGHT/
+                    LIDAR_TOP/
+                    RADAR_BACK_LEFT/
+                    RADAR_BACK_RIGHT/
+                    RADAR_FRONT/
+                    RADAR_FRONT_LEFT/
+                    RADAR_FRONT_RIGHT/
+                v1.0-mini/
+                    attribute.json
+                    calibrated_sensor.json
+                    category.json
+                    ego_pose.json
+                    instance.json
+                    log.json
+                    map.json
+                    sample_annotation.json
+                    sample_data.json
+                    sample.json
+                    scene.json
+                    sensor.json
+                    visibility.json
+            v1.0-test/
+                maps/
+                samples/
+                sweeps/
+                v1.0-test/
+            v1.0-trainval/
+                maps/
+                samples/
+                sweeps/
+                v1.0-trainval/
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.FusionDataset` instance.
+
+    """
+    root_path = os.path.abspath(os.path.expanduser(path))
+
+    dataset = FusionDataset(DATASET_NAME)
+    dataset.notes.is_continuous = True
+    dataset.notes.bin_point_cloud_fields = ["X", "Y", "Z", "Intensity", "Ring"]
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+
+    for index, phase in enumerate(os.listdir(path)):
+        # if phase != "v1.0-mini":
+        #     continue
+        # general_knowledge = _load_general_files(os.path.join(root_path, phase, phase))
+
+        if index == 0:
+            general_knowledge = _load_general_files(os.path.join(root_path, phase, phase))
+
+        phase_path = os.path.join(root_path, phase)
```

Comment:
`os.path.join(root_path, phase)` and `os.path.join(root_path, phase, phase)` have redundant calculation.